site/source/tutorials/10XX_development_environment/index.md

```diff
@@ -0,0 +1,146 @@
+
+In Atom, for example, this means installing [atom-typescript](https://atom.io/packages/atom-typescript). Visual Studio Code comes with typescript support by default, and for Vim try installing a combination of plugins for [language features](https://github.com/Quramy/tsuquyomi) and [syntax highlighting](https://github.com/leafgarland/typescript-vim).
+
+{% instruction 'Test error highlighting' %}
+
+Within the demo app, open `src/widgets/WorkerContainer.ts`. In it, there should be code to create an instance of the Worker widget on line 21. Add a fake property, e.g.:
+
+```typescript
+const workers = workerData.map((worker, i) => w(Worker, {
+  key: `worker-${i}`,
+  fakeProp: 'foo',
+  ...worker
+}));
+```
+
+Verify that your editor has highlighted `fakeProp` as an error:
+
+<p class="center">![error message in Atom](./resources/atom_ts_error.png)</p>
+
+{% instruction 'Go further: VS Code and tasks' %}
+
+You may skip to the next step if you are not interested in trying out VS Code.
+
+VS Code supports Typescript by default, is written is Typescript, and allows you to [configure tasks](https://code.visualstudio.com/docs/editor/tasks) which are then integrated into the IDE.
+
+To configure tasks:
+
+* Press <kbd>⌘⇧P</kbd>/<kbd>Ctrl⇧P</kbd> to bring up the command list.
+* Start typing <kbd>tasks</kbd> and select the command `Tasks: Configure Task Runner`
+
+<p class="center">![gif of opening tasks.json](./resources/task_setup.gif)</p>
+
+The [first tutorial](../000_local_installation/) had you run `dojo build` as the first Dojo CLI command after creating your app, so consider the following code to configure it within `tasks.json`:
+
+```json
+{
+  "version": "0.1.0",
+  "command": "dojo",
+  "isShellCommand": true,
+  "args": [],
+  "showOutput": "always",
+  "suppressTaskName": true,
+  "tasks": [
+    {
+      "taskName": "build",
+      "args": [ "build" ],
+      "isBuildCommand": true,
+      "problemMatcher": {
+      "owner": "dojo",
+      "fileLocation": "relative",
+      "pattern": [
+        {
+          "regexp": "^(\\S+) in (.*)",
+          "severity": 1,
+          "file": 2
+        }, {
+          "regexp": "\\((\\d+),(\\d+)\\):(.*)",
+          "line": 1,
+          "column": 2,
+          "message": 3
+        }
+      ]
+      }
+    }
+  ]
+}
+```
+
+Other tasks to consider adding could be `test` or `watch`. Once configured, the build command (<kbd>⌘⇧B</kbd>/<kbd>Ctrl⇧B</kbd>) will be configured as well as you will have the other
+tasks available in the IDE:
+
+<p class="center">![dropdown of vs code tasks](./resources/run_task.png)</p>
+
+{% section %}
+
+## Debugging
+
+If you are developing a Dojo 2 application in TypeScript, your application code is transpiled from TypeScript to JavaScript. If you are also using dojo build, then your code is also being bundled and minimized. This could make it challenging to debug an application, but we have tried to make Dojo 2 applications integrate well into modern debugging tools.
```

Comment:
This could make it challenging to debug an application, but we have tried to make Dojo 2 applications integrate well into modern debugging tools. -> Bundling and minimizing could make it challenging to debug your application, but we have made efforts to integrate Dojo 2 into modern debugging tools.